### ISSUE TITLE
False positive ParadoxicalCondition seen when variable is assigned with list()

### ISSUE THREAD

**Issue by TysonAndre** [vimeo/psalm#201]
False positive ParadoxicalCondition reproduction case is below (unrelated lines were removed)

Seen with vimeo/psalm 0.3.53 (and master?)

It was false before the assignment, but should be `bool` after that

```php
<?php
class mysync {

    public static function sync(bool $offsetDeltaEnd, $limit = 100) : void {
        // paginate friends if there is a valid offset
        if (!$offsetDeltaEnd) {
            list($offsetDeltaEnd) = self::_hasMore(0, $limit);
            if ($offsetDeltaEnd) {  // Incorrectly warns that this is a paradox
                echo "A";
            } else {
                echo "B";
            }
        }
    }   

    /** @return bool[] */
    public static function _hasMore(int $offset, int $limit) {
        $end = rand() % 2 > 0;
        return [$end];
    }   
}
```

**Comment by muglug** [comment]
Thanks for this as well! The test case above was more than I needed, but it works [online](https://getpsalm.org)
